### PR TITLE
perf_hv_gpci_interface.py: Adding hv_gpci interface events

### PR DESCRIPTION
**Pull request by Tejas3772**
Adding additional perf hv_gpci testcase, for files and access for system P10 or above.

### CODE REVIEW

File: perf/perf_hv_gpci_interface.py.data/perf_hv_gpci_interface.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+hmc_username: 'hscroot'
+hmc_pwd: 'Newbeginning4all'
```
Comment (disgoel) [inline perf/perf_hv_gpci_interface.py.data/perf_hv_gpci_interface.yaml:2]: please remove credentials from public github

File: perf/perf_hv_gpci_interface.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split(':')[-1].strip()
+        return ''
+
+    def run_cmd(self, cmd, user=None):
+        if user == 'test':  # i.e., we are running as root
```
Comment (disgoel) [inline perf/perf_hv_gpci_interface.py:115]: can you correct the comment for test user

**Review by disgoel — CHANGES_REQUESTED** [review]

File: perf/perf_hv_gpci_interface.py (hunk 2 above)
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:115]: done

File: perf/perf_hv_gpci_interface.py.data/perf_hv_gpci_interface.yaml (hunk 1 above)
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py.data/perf_hv_gpci_interface.yaml:2]: done

**Comment by disgoel** [comment]
LGTM

File: perf/perf_hv_gpci_interface.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                read_access = self.run_cmd(f'cat {filepath}', user=user)
+                if read_access is None:
+                    self.fail(f"Read access to {filename} failed as {'root' if user is None else user}.")
+
+                # Attempt write access
+                output = self.run_cmd(f'echo Test > {filepath}', user=user)
+                if 'Permission denied' not in output:
+                    self.fail(f"{'root' if user is None else user} user able to write to {filename}")
+
+    def _disable_and_check_read_access(self):
```
[the remark this answers is not in the document]
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:137]: done

[the remark this answers is not in the document]
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:137]: done

File: perf/perf_hv_gpci_interface.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        dis_performance_cmd = f"chsyscfg -m {self.server} -r lpar -i 'name={self.lpar},allow_perf_collection=0'"
```
[the remark this answers is not in the document]
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:146]: filtered through autopep8 now.

File: perf/perf_hv_gpci_interface.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        self.session.cmd(dis_performance_cmd).stdout_text
+        performance_cmd = f"lssyscfg -m {self.server} -r lpar --filter lpar_names={self.lpar} -F allow_perf_collection"
```
[the remark this answers is not in the document]
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:148]: done sir

File: perf/perf_hv_gpci_interface.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        self.expected_files = self.params.get("access_files", default=None)
+        self.hmc_ip = self.get_mcp_component("HMCIPAddr")
+        if not self.hmc_ip:
+            self.cancel("HMC IP not got")
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.lpar = self.get_partition_name("Partition Name")
```
[the remark this answers is not in the document]
Comment (Tejas3772) [reply perf/perf_hv_gpci_interface.py:54]: I have tried running it on multiple lpars, it is working fine for now.

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**